src/test/scala/dsptools/DspContextSpec.scala

```diff
@@ -49,4 +52,71 @@ class DspContextSpec extends FreeSpec with Matchers {
       DspContext.current.numBits should be (DspContext.defaultNumBits)
     }
   }
+
+  "Test proper nesting of DspContext over module instantiation" in {
+    dsptools.Driver.execute(
+      () => new ContextNestingTop(UInt(4.W), UInt(5.W)),
+      Array("--backend-name", "firrtl")
+    ) { c =>
+      new ContextNestingTester(c)
+    } should be(true)  }
+}
+
+class ContextNestingTester(c: ContextNestingTop[UInt]) extends DspTester(c) {
+  poke(c.io.in1, 15.0)
+  poke(c.io.in2, 2.0)
+
+  expect(c.io.mod1Default, 1.0)
+  expect(c.io.mod1Wrap, 1.0)
+  expect(c.io.mod1Grow, 17.0)
+  expect(c.io.mod2Default, 17.0)
+  expect(c.io.mod2Wrap, 1.0)
+  expect(c.io.mod2Grow, 17.0)
+}
+
+class ContextNestingBottom[T <: Data : Ring](gen1: T, gen2: T) extends Module {
+  val io = IO( new Bundle {
+    val in1 = Input(gen1)
+    val in2 = Input(gen1)
+    val outDefault = Output(gen2)
+    val outWrap    = Output(gen2)
+    val outGrow    = Output(gen2)
+  })
+
+  DspContext.withOverflowType(Wrap) {
+    io.outWrap := io.in1 + io.in2
+  }
+  DspContext.withOverflowType(Grow) {
+    io.outGrow := io.in1 + io.in2
+  }
+
+  io.outDefault := io.in1 + io.in2
+}
+
+class ContextNestingTop[T <: Data : Ring](gen1: T, gen2: T) extends Module {
```

Comment:
@grebe @shunshou If I change this to ```[T <: UInt : Ring]``` the tests here fail because it doesn't end up using the typeclass stuff and thus does not implement the DspContext specified.  This seems dangerous and more than a little hard to describe.  Any ways we can make this more robust